pire/stub/stl.h

```diff
@@ -51,6 +51,11 @@
 #define YASSERT(e) do {} while (0)
 #endif
 
+#ifdef __GNUC__
```

Comment:
Better check for minimal gcc version that supports "diagnostic push", since ancient gcc's like 4.4 don't.